geocoder/src/main/scala/grasshopper/GrasshopperGeocoder.scala

```diff
@@ -28,15 +30,30 @@ object GrasshopperGeocoder extends App with HttpService with WebsocketService {
   lazy val port = config.getString("grasshopper.geocoder.elasticsearch.port")
   lazy val cluster = config.getString("grasshopper.geocoder.elasticsearch.cluster")
 
-  lazy val settings = ImmutableSettings.settingsBuilder()
+  lazy val settings = Settings.settingsBuilder()
     .put("http.enabled", false)
     .put("node.data", false)
     .put("node.master", false)
     .put("cluster.name", cluster)
     .put("client.transport.sniff", true)
 
-  override lazy val client = new TransportClient(settings)
-    .addTransportAddress(new InetSocketTransportAddress(host, port.toInt))
+  lazy val clientBuilder = TransportClient.builder()
+
+  lazy val user = config.getString("grasshopper.geocoder.elasticsearch.user")
+  lazy val password = config.getString("grasshopper.geocoder.elasticsearch.password")
+
+  if (!user.isEmpty && !password.isEmpty) {
```

Comment:
May I suggest using `if (user.nonEmpty && password.nonEmpty)` here?. I think it reads better